service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseSubmissionValidatorService.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+@Service
+public class CaseSubmissionValidatorService {
+
+    @Autowired
+    public CaseSubmissionValidatorService() {
+
+    }
+
+    public List<String> validateCaseDetails(CaseData caseData) {
+        ImmutableList.Builder<String> caseErrors = ImmutableList.builder();
+
+        Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+        Set<ConstraintViolation<CaseData>> violations = validator.validate(caseData);
+
+        List<String> errors = Stream.of("applicant", "children", "orders", "groundsForTheApplication", "hearing",
+            "caseName", "documents")
+            .flatMap(section -> Stream.of(buildErrorSection(violations, section)))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+
+        errors.forEach(error -> caseErrors.add(error));
```

Comment:
Consider:

      ```
  Stream.of("applicant", "children", "orders", "groundsForTheApplication", "hearing",
            "caseName", "documents")
            .flatMap(section -> Stream.of(buildErrorSection(violations, section)))
            .flatMap(Collection::stream)
            .forEach(caseErrors::add);
```
Collect and then forEach seems superfluous - you are going through the whole collection twice when it is not necessary.